pkg/generate/app/app.go

```diff
@@ -81,6 +85,28 @@ func nameFromGitURL(url *url.URL) (string, bool) {
 	}
 	// from path
 	if name, ok := git.NameFromRepositoryURL(url); ok {
+
+		if !kvalidation.IsDNS1123Label(name) {
+			name = strings.ToLower(name)
+
+			// remove everything except [-0-9a-z]
+			name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
```

Comment:
There should be a function somewhere else that already does this, if not it would be better to have it in one common place in generation.  However, I'm not totally sure that this belongs down at this level.  The code that uses the name suggestion should probably apply this logic, instead of within this segment.